include/revng-c/RestructureCFG/CycleEquivalenceClass.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+//
+// Copyright rev.ng Labs Srl. See LICENSE.md for details.
+//
+
+#include <cstddef>
+
+#include "llvm/ADT/SmallSet.h"
+
+#include "revng/Support/Assert.h"
+#include "revng/Support/Debug.h"
+#include "revng/Support/GraphAlgorithms.h"
+
+template<class NodeT>
+class CycleEquivalenceClass {
+public:
+  using EdgeDescriptor = revng::detail::EdgeDescriptor<NodeT>;
+
+  using edge_container = llvm::SmallSet<EdgeDescriptor, 4>;
+  using edge_const_iterator = edge_container::const_iterator;
+
+private:
+  size_t ID;
+  edge_container Edges;
+
+public:
+  CycleEquivalenceClass(size_t ID) : ID(ID) {}
+
+  size_t getID();
+
+  void insertEdge(EdgeDescriptor &E);
```

Comment:
Just `insert`?

Reply:
Changed this here https://github.com/revng/revng-c/blob/323b04da3a0979143e3207968b19e7142893109e/include/revng-c/RestructureCFG/CycleEquivalenceClass.h#L31